rslib/src/scheduler/answering/mod.rs

```diff
@@ -148,7 +148,10 @@ impl CardStateUpdater {
                 match filtered {
                     FilteredState::Preview(next) => self.apply_preview_state(current, next),
                     FilteredState::Rescheduling(next) => {
-                        self.apply_normal_study_state(current, next.original_state)
+                        let revlog = self.apply_normal_study_state(current, next.original_state);
+                        self.card.original_due = self.card.due;
```

Comment:
I suspect I'm missing something - why is this line required? I would have thought this would already be handled when the card is moved into the filtered deck.

Reply:
[This assert](https://github.com/ankitects/anki/blob/312d396505a4806b76b74dfda6d276ec595e17ed/pylib/tests/test_schedv3.py#L718) fails without it. That's because remove_from_filtered_deck_restoring_queue() is restoring the original due date before the card was moved to the filtered deck, regardless of whether rescheduling is enabled.

Reply:
Ah, thanks, that makes sense.